docs/src/content/docs/cookbook/dexes/dedust.mdx

```diff
@@ -7,8 +7,430 @@ sidebar:
 
-:::danger[Not implemented]
+Before going further, familiarize yourself with the following:
 
-  This page is a stub until it gets new content in [#146](https://github.com/tact-lang/tact-docs/issues/146).
+* [Receiving messages](/book/receive/)
+* [Sending messages](/book/send/)
+* [Fungible Tokens (Jettons)](/cookbook/jettons/)
+* [DeDust Docs: Concepts](https://docs.dedust.io/docs/concepts)
+
+## Swaps
+
+Read more about swaps in the [DeDust documentation](https://docs.dedust.io/docs/swaps).
+
+:::caution
+
+  It's important to ensure that contracts are deployed. Sending funds to an inactive contract could result in irretrievable loss.
+
+:::
+
+All kinds of swaps use `SwapStep{:tact}` and `SwapParams{:tact}` structures:
+
+```tact
+/// https://docs.dedust.io/reference/tlb-schemes#swapstep
+struct SwapStep {
+    // The pool that will do the swapping, i.e. pairs like TON/USDT or USDT/DUST
+    poolAddress: Address;
+
+    // A kind of swap to make, can only be 0 as of now
+    kind: Int as uint1 = 0;
+
+    // Minimum output of the swap
+    // If the actual value is less than specified, the swap will be rejected
+    limit: Int as coins = 0;
+
+    // Reference to the next step, which can be used for multi-hop swaps
+    // The type here is actually `SwapStep?`,
+    // but specifying recursive types isn't allowed in Tact yet
+    nextStep: Cell?;
+}
+
+/// https://docs.dedust.io/reference/tlb-schemes#swapparams
+struct SwapParams {
+    // Specifies a deadline for the swap to reject the swap coming to the pool late
+    // Accepts the number of seconds passed since the UNIX Epoch
+    // Defaults to 0, which removes the deadline
+    deadline: Int as uint32 = 0;
+
+    // Specifies an address where funds will be sent after the swap
+    // Defaults to `null`, which makes the swap use the sender's address
+    recipientAddress: Address? = null;
+
+    // Referral address, required for the referral program of DeDust
+    // Defaults to `null`
+    referralAddress: Address? = null;
+
+    // Custom payload that will be attached to the fund transfer upon a successful swap
+    // Defaults to `null`
+    fulfillPayload: Cell? = null;
+
+    // Custom payload that will be attached to the fund transfer upon a rejected swap
+    // Defaults to `null`
+    rejectPayload: Cell? = null;
+}
+```
+
+### Swap Toncoin for any Jetton
+
+```tact
+/// https://docs.dedust.io/reference/tlb-schemes#message-swap
+message(0xea06185d) NativeSwap {
+    // Unique identifier used to trace transactions across multiple contracts and to prevent replay attacks
+    // Defaults to 0, which means Native Vault will handle this for us
+    queryId: Int as uint64 = 0;
+
+    // Toncoin amount for the swap
+    amount: Int as coins;
+
+    // Inlined fields of SwapStep Struct
```

Comment:
Sounds interesting! @Gusarich can you please open an issue on this?

Reply:
@anton-trunov actually after checking how nested structs work it turned out that they are already being inlined by default, and only put in a ref if they don't fit in current cell. So probably @a-bahdanau can rewrite some struct definitions here

Reply:
@Gusarich just as @anton-trunov said, we cannot enforce inlining as of now. I did test the code and Tact did inline SwapStep Struct automagically since it fit into the Cell, but ultimately decided not to because of lack of guarantees.

But the idea of `as inline` is good, and we should also consider `as ref` for explicitly placing the Struct/Message contents in a ref cell

Reply:
https://github.com/tact-lang/tact/issues/1050

Reply:
I'll mark this as resolved because this wasn't even a change request for this PR, just a related comment.